File: CS/CustomBarSubItem/CustomBarManager/CustomBarSubItem.cs
using System;
using System.Collections.Generic;
using System.Linq;
using DevExpress.XtraBars;

namespace CustomBar.CustomBarManager {
    class CustomBarSubItem : BarSubItem{

        public const string BaseBarItemName = "BarSubItem";
        public const string BarItemName = "CustomBarCheckSubItem";
        public const string SubBarItemCaption = "CustomBarCheckSubItem";

        private bool isChecked;

        public bool IsChildItemChecked {
            get {
                return isChecked;
            }
            set {
                if (value != isChecked) 
                { 
                    isChecked = value; 
                }
            }
        }
        
        public CustomBarSubItem(BarManager manager, string caption)
            : base(manager, caption) {
                IsChildItemChecked = false;
                
        }

        public CustomBarSubItem(BarManager manager, string caption, BarItem[] items)
            : base(manager, caption, items) {
            
        }

        public CustomBarSubItem(BarManager manager, string caption, int imageIndex)
            : base(manager, caption, imageIndex) {
            
        }

        public CustomBarSubItem(BarManager manager, string caption, int imageIndex, BarItem[] items)
            : base(manager, caption, imageIndex, items) {
            
        }
    }   
}

File: VB/CustomBarSubItem/CustomBarManager/CustomBarSubItem.vb
Imports Microsoft.VisualBasic
Imports System
Imports System.Collections.Generic
Imports System.Linq
Imports DevExpress.XtraBars

Namespace CustomBar.CustomBarManager
	Friend Class CustomBarSubItem
		Inherits BarSubItem

		Public Const BaseBarItemName As String = "BarSubItem"
		Public Const BarItemName As String = "CustomBarCheckSubItem"
		Public Const SubBarItemCaption As String = "CustomBarCheckSubItem"

		Private isChecked As Boolean

		Public Property IsChildItemChecked() As Boolean
			Get
				Return isChecked
			End Get
			Set(ByVal value As Boolean)
				If value <> isChecked Then
					isChecked = value
				End If
			End Set
		End Property

		Public Sub New(ByVal manager As BarManager, ByVal caption As String)
			MyBase.New(manager, caption)
				IsChildItemChecked = False

		End Sub

		Public Sub New(ByVal manager As BarManager, ByVal caption As String, ByVal items() As BarItem)
			MyBase.New(manager, caption, items)

		End Sub

		Public Sub New(ByVal manager As BarManager, ByVal caption As String, ByVal imageIndex As Integer)
			MyBase.New(manager, caption, imageIndex)

		End Sub

		Public Sub New(ByVal manager As BarManager, ByVal caption As String, ByVal imageIndex As Integer, ByVal items() As BarItem)
			MyBase.New(manager, caption, imageIndex, items)

		End Sub
	End Class
End Namespace
